### PR TITLE
[AzureMonitorExporter] refactor demo projects

### PR DESCRIPTION
**Pull request by TimothyMothra**
Refactor demo project as was discussed here: https://github.com/Azure/azure-sdk-for-net/pull/30860#discussion_r965065100

### Changes
- Demo Project
  - created sub directories: "Logs", "Metrics", and "Traces".
    Future work: customization examples will be added to these directories to accompany public Documentation.
  - broke up the previous `Program.cs` into three new classes "LogDemo", "MetricDemo", and "TraceDemo".
- Traces.Customization project
  - removed project
  - moved `ActivityEnrichingProcessor` and `ActivityFilteringProcessor` to "Demo\Traces" directory.

### CODE REVIEW

**Review by vishweshbankwar — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Metrics/MetricDemo.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            MyFruitCounter.Add(1, new("name", "lemon"), new("color", "yellow"));
+            MyFruitCounter.Add(2, new("name", "apple"), new("color", "green"));
+            MyFruitCounter.Add(5, new("name", "apple"), new("color", "red"));
+            MyFruitCounter.Add(4, new("name", "lemon"), new("color", "yellow"));
+        }
+
+        public void Dispose()
+        {
+            this.meterProvider.Shutdown();
+            this.meterProvider.Dispose();
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Metrics/MetricDemo.cs:41]: Do we need both `Shutdown` and `Dispose` here?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Traces/TraceDemo.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                {
+                    nestedActivity?.SetTag("bar", "Hello, World!");
+                    nestedActivity?.SetStatus(ActivityStatusCode.Ok);
+                }
+            }
+        }
+
+        public void Dispose()
+        {
+            this.tracerProvider.Shutdown();
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Traces/TraceDemo.cs:69]: Looked at source in OTel, calling Dispose should be enough here.

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM, left minor suggestion on `shutdown`/`dispose`.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Traces/TraceDemo.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            var resourceAttributes = new Dictionary<string, object>
+            {
+                { "service.name", "my-service" },
+                { "service.namespace", "my-namespace" },
+                { "service.instance.id", "my-instance" },
+            };
+
+            var resourceBuilder = ResourceBuilder.CreateDefault().AddAttributes(resourceAttributes);
+
+            this.tracerProvider = Sdk.CreateTracerProviderBuilder()
```
Comment (samsp-msft) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Traces/TraceDemo.cs:33]: Its confusing that the ActivityEnrichingProcessor() and ActivityFilteringProcessor() are part of this project and not the SDK. Given that there are links to just this file as a sample, I got lost trying to find these - especially as github search is somewhat broken these days,

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Traces/TraceDemo.cs:33]: HI @samsp-msft, point taken. I'll bring this up with the team about making these more discoverable. Thanks!

**Pull request merged**